### PR TITLE
Add ESLint to API

### PR DESCRIPTION
**Pull request by leite08**
Ref https://github.com/metriport/metriport-internal/issues/143

### Dependencies

- Upstream: https://github.com/metriport/metriport/pull/57
- Downstream: https://github.com/metriport/metriport/pull/59

### Description

- Add ESLint to API
- Fix linting issues

### Release Plan

- anytime

### CODE REVIEW

**Comment by leite08** [comment]
See https://github.com/metriport/metriport-internal/pull/260

**Review by Goncharo — APPROVED** [review]
Few comments, good to 🚢  after addressed as needed

File: api/app/src/command/webhook/webhook.ts [hunk 1]
```diff
@@ -98,7 +97,7 @@ export const processData = async <T extends MetriportData>(data: UserData<T>[]):
           // transform each chunk into a payload
           const payloads = chunks.map(c => {
             // groups by user
-            const dataByUser = groupBy(dataAndUserList, v => v.cxUserId);
+            const dataByUser = groupBy(c, v => v.cxUserId);
```
[the remark this answers is not in the document]
Comment (leite08) [reply api/app/src/command/webhook/webhook.ts:100]: Good catch! I was looking for it to make it clear.

Yes, I noticed the bug bc the compiler complained of unused variable, `c`. Reading the code we can see we're not using the chunk's data to group by user, but the whole list.

File: api/app/src/domain/user-token.ts [hunk 2]
```diff
@@ -5,9 +5,13 @@ import { Util } from "../shared/util";
 
 export type UserTokenCreate = OmitProperties<
   Omit<UserToken, "token" | "expiryTime"> & Partial<Pick<UserToken, "token" | "expiryTime">>,
+  // Only safe bc we're not relyint on the function signature, just omiting all functions ferm the resulting type
+  // eslint-disable-next-line @typescript-eslint/ban-types
   Function
 >;
 
+// Only safe bc we're not relyint on the function signature, just omiting all functions ferm the resulting type
```
[the remark this answers is not in the document]
Comment (leite08) [reply api/app/src/domain/user-token.ts:13]: 🕵️ thanks!

File: api/app/src/providers/oauth2.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Axios, AxiosResponse } from "axios";
+import axios, { AxiosResponse } from "axios";
```
[the remark this answers is not in the document]
Comment (leite08) [reply api/app/src/providers/oauth2.ts:1]: yeah, I did a test here running the Widget against the API and running OAuth v1. We don't need to create a global instance - I think its mostly useful in the case to reuse a specific configuration. Not sure about performance, though.

Had to change from `require` as ESLint was complaining about it along with `import` - so went with the simplest ES6 version.

**Review by Goncharo — APPROVED** [review]
FYI there are some merge conflicts, otherwise 🚢 🚢 🚢 🚢

**Pull request merged**